lib/chef/resource/aws_security_group.rb

```diff
@@ -49,8 +49,11 @@ class Chef::Resource::AwsSecurityGroup < Chef::Provisioning::AWSDriver::AWSResou
   }
 
   def aws_object
-    driver, id = get_driver_and_id
-    result = driver.ec2.security_groups[id] if id
+    if security_group_id
+      result = driver.ec2.security_groups[security_group_id]
+    else
+      result = driver.ec2.security_groups.filter('group-name', name).first
```

Comment:
TODO if more than 1 comes back, we need to filter by VPC.  If the user does not provide VPC we blow up and say why.